Regression tests failing

```
======================================================================
FAIL: runTest (__main__.MultipolygonSlimTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression-test.py", line 461, in runTest
    self.executeStatements(self.postDiffStatements)
  File "tests/regression-test.py", line 362, in executeStatements
    str(res) + "  -- " + sql_test_statements[i][1] + " (" + sql_test_statements[i][2] + ") {" + str(self.parameters) +"}")
AssertionError: 34: Sql statement returned more than one result: [(12893.0,), (12893.0,)]  -- Multipolygon non-area inner - outer (Tags from relation) (SELECT round(ST_Area(way)) FROM planet_osm_polygon WHERE osm_id = -14 and landuse = 'residential' and name = 'Name_way5') {[]}

======================================================================
FAIL: runTest (__main__.MultipolygonSlimTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression-test.py", line 461, in runTest
    self.executeStatements(self.postDiffStatements)
  File "tests/regression-test.py", line 362, in executeStatements
    str(res) + "  -- " + sql_test_statements[i][1] + " (" + sql_test_statements[i][2] + ") {" + str(self.parameters) +"}")
AssertionError: 34: Sql statement returned more than one result: [(12893.0,), (12893.0,)]  -- Multipolygon non-area inner - outer (Tags from relation) (SELECT round(ST_Area(way)) FROM planet_osm_polygon WHERE osm_id = -14 and landuse = 'residential' and name = 'Name_way5') {['-G']}

======================================================================
FAIL: runTest (__main__.MultipolygonSlimTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression-test.py", line 461, in runTest
    self.executeStatements(self.postDiffStatements)
  File "tests/regression-test.py", line 362, in executeStatements
    str(res) + "  -- " + sql_test_statements[i][1] + " (" + sql_test_statements[i][2] + ") {" + str(self.parameters) +"}")
AssertionError: 34: Sql statement returned more than one result: [(12893.0,), (12893.0,)]  -- Multipolygon non-area inner - outer (Tags from relation) (SELECT round(ST_Area(way)) FROM planet_osm_polygon WHERE osm_id = -14 and landuse = 'residential' and name = 'Name_way5') {['-k']}

======================================================================
FAIL: runTest (__main__.MultipolygonSlimTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression-test.py", line 461, in runTest
    self.executeStatements(self.postDiffStatements)
  File "tests/regression-test.py", line 362, in executeStatements
    str(res) + "  -- " + sql_test_statements[i][1] + " (" + sql_test_statements[i][2] + ") {" + str(self.parameters) +"}")
AssertionError: 34: Sql statement returned more than one result: [(12893.0,), (12893.0,)]  -- Multipolygon non-area inner - outer (Tags from relation) (SELECT round(ST_Area(way)) FROM planet_osm_polygon WHERE osm_id = -14 and landuse = 'residential' and name = 'Name_way5') {['-x', '-k', '--hstore-match-only']}

======================================================================
FAIL: runTest (__main__.MultipolygonSlimTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression-test.py", line 461, in runTest
    self.executeStatements(self.postDiffStatements)
  File "tests/regression-test.py", line 362, in executeStatements
    str(res) + "  -- " + sql_test_statements[i][1] + " (" + sql_test_statements[i][2] + ") {" + str(self.parameters) +"}")
AssertionError: 75: Sql statement returned more than one result: [(15246.0,), (15246.0,)]  -- Multipolygon tags on both inner and outer diff change outer (presence of relation) (SELECT round(ST_Area(way)) FROM planet_osm_polygon WHERE osm_id = -34 and "landuse" = 'farmland') {['--tag-transform-script', 'style.lua', '-k']}

----------------------------------------------------------------------
```

bisecting finds it broke in aab01ddd65cfc1879

--- a/id-tracker.cpp
+++ b/id-tracker.cpp
@@ -171,6 +171,8 @@ osmid_t id_tracker::pop_mark() {
 
 size_t id_tracker::size() { return impl->count; }
 
+osmid_t id_tracker::last_returned() const { return impl->old_id; }
+
 bool id_tracker::is_valid(osmid_t id) { return id != max(); }
 osmid_t id_tracker::max() { return std::numeric_limits<osmid_t>::max(); }
 osmid_t id_tracker::min() { return std::numeric_limits<osmid_t>::min(); }

--- a/id-tracker.hpp
+++ b/id-tracker.hpp
@@ -13,6 +13,7 @@ struct id_tracker : public boost::noncopyable {
     bool is_marked(osmid_t id);
     osmid_t pop_mark();
     size_t size();
+    osmid_t last_returned() const;
 
     static bool is_valid(osmid_t);
     static osmid_t max();

--- a/output-multi.cpp
+++ b/output-multi.cpp
@@ -61,6 +61,15 @@ size_t output_multi_t::pending_count() const {
 }
 
 void output_multi_t::enqueue_ways(pending_queue_t &job_queue, osmid_t id, size_t output_id, size_t& added) {
+    osmid_t const prev = ways_pending_tracker->last_returned();
+    if (id_tracker::is_valid(prev) && prev >= id) {
+        if (prev > id) {
+            job_queue.push(pending_job_t(id, output_id));
+        }
+        // already done the job
+        return;
+    }
+
     //make sure we get the one passed in
     if(!ways_done_tracker->is_marked(id) && id_tracker::is_valid(id)) {
         job_queue.push(pending_job_t(id, output_id));
@@ -83,11 +92,10 @@ void output_multi_t::enqueue_ways(pending_queue_t &job_queue, osmid_t id, size_t
 
     //make sure to get this one as well and move to the next
     if(popped == id) {
-        popped = ways_pending_tracker->pop_mark();
-    }
-    if (!ways_done_tracker->is_marked(popped) && id_tracker::is_valid(popped)) {
-        job_queue.push(pending_job_t(popped, output_id));
-        added++;
+        if (!ways_done_tracker->is_marked(popped) && id_tracker::is_valid(popped)) {
+            job_queue.push(pending_job_t(popped, output_id));
+            added++;
+        }
     }
 }
 
@@ -106,6 +114,15 @@ int output_multi_t::pending_way(osmid_t id, int exists) {
 }
 
 void output_multi_t::enqueue_relations(pending_queue_t &job_queue, osmid_t id, size_t output_id, size_t& added) {
+    osmid_t const prev = rels_pending_tracker->last_returned();
+    if (id_tracker::is_valid(prev) && prev >= id) {
+        if (prev > id) {
+            job_queue.push(pending_job_t(id, output_id));
+        }
+        // already done the job
+        return;
+    }
+
     //make sure we get the one passed in
     if(id_tracker::is_valid(id)) {
         job_queue.push(pending_job_t(id, output_id));
@@ -126,11 +143,10 @@ void output_multi_t::enqueue_relations(pending_queue_t &job_queue, osmid_t id, s
 
     //make sure to get this one as well and move to the next
     if(popped == id) {
-        popped = rels_pending_tracker->pop_mark();
-    }
-    if(id_tracker::is_valid(popped)) {
-        job_queue.push(pending_job_t(popped, output_id));
-        added++;
+        if(id_tracker::is_valid(popped)) {
+            job_queue.push(pending_job_t(popped, output_id));
+            added++;
+        }
     }
 }
 

--- a/output-pgsql.cpp
+++ b/output-pgsql.cpp
@@ -273,6 +273,15 @@ extern "C" void *pthread_output_pgsql_stop_one(void *arg) {
 } // anonymous namespace
 
 void output_pgsql_t::enqueue_ways(pending_queue_t &job_queue, osmid_t id, size_t output_id, size_t& added) {
+    osmid_t const prev = ways_pending_tracker->last_returned();
+    if (id_tracker::is_valid(prev) && prev >= id) {
+        if (prev > id) {
+            job_queue.push(pending_job_t(id, output_id));
+        }
+        // already done the job
+        return;
+    }
+
     //make sure we get the one passed in
     if(!ways_done_tracker->is_marked(id) && id_tracker::is_valid(id)) {
         job_queue.push(pending_job_t(id, output_id));
@@ -294,12 +303,11 @@ void output_pgsql_t::enqueue_ways(pending_queue_t &job_queue, osmid_t id, size_t
     }
 
     //make sure to get this one as well and move to the next
-    if(popped == id) {
-        popped = ways_pending_tracker->pop_mark();
-    }
-    if (!ways_done_tracker->is_marked(popped) && id_tracker::is_valid(popped)) {
-        job_queue.push(pending_job_t(popped, output_id));
-        added++;
+    if(popped > id) {
+        if (!ways_done_tracker->is_marked(popped) && id_tracker::is_valid(popped)) {
+            job_queue.push(pending_job_t(popped, output_id));
+            added++;
+        }
     }
 }
 
@@ -319,6 +327,15 @@ int output_pgsql_t::pending_way(osmid_t id, int exists) {
 }
 
 void output_pgsql_t::enqueue_relations(pending_queue_t &job_queue, osmid_t id, size_t output_id, size_t& added) {
+    osmid_t const prev = rels_pending_tracker->last_returned();
+    if (id_tracker::is_valid(prev) && prev >= id) {
+        if (prev > id) {
+            job_queue.push(pending_job_t(id, output_id));
+        }
+        // already done the job
+        return;
+    }
+
     //make sure we get the one passed in
     if(id_tracker::is_valid(id)) {
         job_queue.push(pending_job_t(id, output_id));
@@ -338,12 +355,11 @@ void output_pgsql_t::enqueue_relations(pending_queue_t &job_queue, osmid_t id, s
     }
 
     //make sure to get this one as well and move to the next
-    if(popped == id) {
-        popped = rels_pending_tracker->pop_mark();
-    }
-    if(id_tracker::is_valid(popped)) {
-        job_queue.push(pending_job_t(popped, output_id));
-        added++;
+    if(popped > id) {
+        if(id_tracker::is_valid(popped)) {
+            job_queue.push(pending_job_t(popped, output_id));
+            added++;
+        }
     }
 }
 

